src/main/java/bio/terra/service/iam/IamService.java

```diff
@@ -27,10 +34,18 @@
 @Component
 public class IamService {
     private final IamProviderInterface iamProvider;
+    private final ConfigurationService configurationService;
+    private Map<AuthorizedCacheKey, AuthorizedCacheValue> authorizedMap;
+    private int cacheSize;
 
     @Autowired
-    public IamService(IamProviderInterface iamProvider) {
+    public IamService(IamProviderInterface iamProvider,
+                      ConfigurationService configurationService) {
         this.iamProvider = iamProvider;
+        this.configurationService = configurationService;
+        cacheSize = configurationService.getParameterValue(AUTH_CACHE_SIZE);
+        // wrap the cache map with a synchronized map to safely share the cache across threads
+        authorizedMap = Collections.synchronizedMap(new LRUMap<>(cacheSize));
```

Comment:
Given that this is used across threads, I think it's extra important to make authorizedMap a final.  Generally, I think it's a great idea to make all ~~member~~ variables final to be safe